monitor/flowgen/bridging/bridging.go

```diff
@@ -96,48 +117,61 @@ func openOrder(
 			Amount: depositWithFee.Amount,
 		},
 		Expenses: []solvernet.Expense{}, // Explicit expense not required for native transfer calls.
-		Calls: []bindings.SolverNetCall{
-			{
-				Target: owner,
-				Value:  expense.Amount,
-			},
-		},
+		Calls:    []bindings.SolverNetCall{call.ToBinding()},
 	}
 
 	orderID, err := solvernet.OpenOrder(ctx, networkID, conf.srcChain, backends, owner, orderData)
 	if err != nil {
 		return types.Result{}, false, errors.Wrap(err, "open order")
 	}
 
+	if err = checker(ctx, stypes.CheckRequest{
```

Comment:
avoid inlining such complex things, it is hard to read, break this into multiple steps